packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -946,21 +984,51 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
   void _cursorTick(Timer timer) {
     _showCursor.value = !_showCursor.value;
+    if (widget.cursorOpacityAnimates) {
+      // If we want to show the cursor, we will animate the opacity to the value
+      // of 1.0, and likewise if we want to make it disappear, to 0.0. An easing
+      // curve is used for the animation to mimic the aesthetics of the native
+      // iOS cursor.
+      final double toValue = _showCursor.value ? 1.0 : 0.0;
+      _cursorController.animateTo(toValue, duration: _fadeDuration, curve: Curves.easeOut);
+    } else {
+      _cursorController.value = _showCursor.value ? 1.0 : 0.0;
+    }
+
     if (_obscureShowCharTicksPending > 0) {
-      setState(() { _obscureShowCharTicksPending--; });
+      setState(() {
+        _obscureShowCharTicksPending--;
+      });
     }
   }
 
+  void _cursorWaitForStart(Timer timer) {
+    _cursorTimer?.cancel();
+    _cursorTimer = Timer.periodic(_kCursorBlinkHalfPeriod, _cursorTick);
```

Comment:
Oh, I get what _kCursorBlinkHalfPeriod is now. 

I think when it's hard to describe, let's stick to industry terms with prior art. Like signal processing or something https://goo.gl/images/n99F1c. Since it's private, let's rename it.